Libraries/AzureAdminClient/AzureAdminClientLib/VmOps.cs

```diff
@@ -1850,7 +1852,75 @@ private HttpResponse RestartVmArm(string vmName, string serviceName)
 
             return resp;
         }
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="vmName"></param>
+        /// <param name="serviceName"></param>
+        /// <param name="deleteFromStorage"></param>
+        /// <param name="throwIfNotFound"></param>
+        /// <returns></returns>
+        public HttpResponse DeleteNic(string networkInterfaceName,
+           string serviceName, bool deleteFromStorage, bool throwIfNotFound)
+        {
+            var requestUrl = string.Format(URLTEMPLATE_DELETENIC_ARM,
+                Connection.SubcriptionID, serviceName, networkInterfaceName, APIVERSION_VMOPS);
+            HttpResponse resp = new HttpResponse();
+            if (networkInterfaceName == null) return resp;
+            for (var index = 0; index < 5; index++)
+            {
+                var hi = new HttpInterface(Connection);
+                resp = hi.PerformRequestArm(HttpInterface.RequestType_Enum.DELETE, requestUrl);
+                if (!resp.HadError)
+                    break;
+                else
+                {
+                    if ((5 > index++))
+                        Thread.Sleep((int)(1000 * Math.Pow(4.17, index + 1)));

```

Comment:
Sweet! We are using this function quite a bit in the code, though. It'd be great if we can have it in a common class such as Utilities or something. Just an enhancement for when you guys have time.